### PR TITLE
*: fix redact log

### LINKED ISSUE #8419: Some key are not replaced in log
### What did you do?

Set enable-redact-log to true

### What did you expect to see?

All key in the log should be replaced with `?`

### What did you see instead?

```
[2024/07/18 11:32:59.548 +08:00] [INFO] [keyspace.go:383] ["[keyspace] added region label for keyspace"] [keyspace-id=0] [label-rule="{\"id\":\"keyspaces/0\",\"index\":0,\"labels\":[{\"key\":\"id\",\"value\":\"0\"}],\"rule_type\":\"key-range\",\"data\":[{\"start_key\":\"7200000000000000fb\",\"end_key\":\"7200000100000000fb\"},{\"start_key\":\"7800000000000000fb\",\"end_key\":\"7800000100000000fb\"}]}"] [takes=1.505851ms]
[2024/07/18 11:32:59.787 +08:00] [INFO] [cluster_worker.go:248] ["region batch split, generate new regions"] [region-id=12] [origin="id:14 end_key:\"748000FFFFFFFFFFFFF900000000000000F8\" region_epoch:<conf_ver:1 version:7 > peers:<id:15 store_id:1 >"] [total=1]
```

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #8419.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Manual test

previous with redact log enabled:
```
[2024/07/18 11:32:59.548 +08:00] [INFO] [keyspace.go:383] ["[keyspace] added region label for keyspace"] [keyspace-id=0] [label-rule="{\"id\":\"keyspaces/0\",\"index\":0,\"labels\":[{\"key\":\"id\",\"value\":\"0\"}],\"rule_type\":\"key-range\",\"data\":[{\"start_key\":\"7200000000000000fb\",\"end_key\":\"7200000100000000fb\"},{\"start_key\":\"7800000000000000fb\",\"end_key\":\"7800000100000000fb\"}]}"] [takes=1.505851ms]
[2024/07/18 11:32:59.787 +08:00] [INFO] [cluster_worker.go:248] ["region batch split, generate new regions"] [region-id=12] [origin="id:14 end_key:\"748000FFFFFFFFFFFFF900000000000000F8\" region_epoch:<conf_ver:1 version:7 > peers:<id:15 store_id:1 >"] [total=1]
```

now with redact log disabled:
```
[2024/07/18 11:24:55.806 +08:00] [INFO] [keyspace.go:384] ["[keyspace] added region label for keyspace"] [keyspace-id=0] [label-rule="id: keyspaces/0, index: 0, type: key-range, labels: [key: id, value: 0], data: [startKey: {7200000000000000fb}, endKey: {7200000100000000fb}, startKey: {7800000000000000fb}, endKey: {7800000100000000fb}]"] [takes=1.031017ms]
[2024/07/18 11:24:56.768 +08:00] [INFO] [cluster_worker.go:248] ["region batch split, generate new regions"] [region-id=10] [origin="id:12 end_key:\"748000FFFFFFFFFFFFFA00000000000000F8\" region_epoch:<conf_ver:1 version:6 > peers:<id:13 store_id:1 >"] [total=1]
```

now with redact log enabled:
```
[2024/07/18 11:26:17.840 +08:00] [INFO] [keyspace.go:384] ["[keyspace] added region label for keyspace"] [keyspace-id=0] [label-rule=?] [takes=1.007138ms]
[2024/07/18 11:26:20.371 +08:00] [INFO] [cluster_worker.go:248] ["region batch split, generate new regions"] [region-id=14] [origin=?] [total=1]
```

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by okJiang** [comment]
Where can we find out what content needs to be redacted?

**Review by okJiang — APPROVED** [review]

**Comment by rleungx** [comment]
> Where can we find out what content needs to be redacted?

Mostly, it is the region's key range. PD doesn't have any other information related to the user data.

File: pkg/schedule/labeler/rules.go [hunk 1]
```diff
@@ -38,6 +39,10 @@ type RegionLabel struct {
 	expire  *time.Time
 }
 
+func (l *RegionLabel) String() string {
+	return fmt.Sprintf("key: %s, value: %s", l.Key, l.Value)
```
Comment (HuSharp) [inline pkg/schedule/labeler/rules.go:43]: Does it need to be `{}` as well?
```suggestion
	return fmt.Sprintf("key: {%s}, value: {%s}", l.Key, l.Value)
```

File: pkg/schedule/labeler/rules.go [hunk 2]
```diff
@@ -78,6 +114,10 @@ type KeyRangeRule struct {
 	EndKeyHex   string `json:"end_key"`   // hex format end key, for marshal/unmarshal
 }
 
+func (r *KeyRangeRule) String() string {
+	return fmt.Sprintf("startKey: %s, endKey: %s", r.StartKeyHex, r.EndKeyHex)
```
Comment (HuSharp) [inline pkg/schedule/labeler/rules.go:118]: ditto

File: pkg/schedule/labeler/rules.go (hunk 1 above)
Comment (rleungx) [reply pkg/schedule/labeler/rules.go:43]: No

File: pkg/schedule/labeler/rules.go (hunk 2 above)
Comment (rleungx) [reply pkg/schedule/labeler/rules.go]: Oh, this function is no longer needed.

File: pkg/schedule/labeler/rules.go (hunk 1 above)
Comment (HuSharp) [reply pkg/schedule/labeler/rules.go:43]: got it

**Review by HuSharp — APPROVED** [review]

**Review by HuSharp — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**